src/commands/deploy.ts

```diff
@@ -120,7 +120,22 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         const warning: string = `Are you sure you want to deploy to "${node.treeItem.client.fullName}"? This will overwrite any previous deployment and cannot be undone.`;
         context.properties.cancelStep = 'confirmDestructiveDeployment';
         const deployButton: vscode.MessageItem = { title: 'Deploy' };
-        await ext.ui.showWarningMessage(warning, { modal: true }, deployButton, DialogResponses.cancel);
+        if (defaultWebAppToDeploy) {
+            const changeDefault: vscode.MessageItem = { title: 'Change default' };
+            const result: vscode.MessageItem = await ext.ui.showWarningMessage(warning, { modal: true }, deployButton, DialogResponses.cancel, changeDefault);
+            if (result.title === 'Change default') {
+                const localRootPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.path : "";
```

Comment:
Rather than getting the `localRootPath` again, you can check the `currentWorkspace` variable above:
l45: `currentWorkspace = vscode.workspace.workspaceFolders[0];`

If it's undefined, you know that you don't have to worry about deployConfig.